net/davix/src/TDavixFile.cxx

```diff
@@ -736,11 +737,11 @@ void TDavixFile::Seek(Long64_t offset, ERelativeTo pos)
 
 Bool_t TDavixFile::ReadBuffer(char *buf, Int_t len)
 {
-   TLockGuard guard(&(d_ptr->positionLock));
    Davix_fd *fd;
    if ((fd = d_ptr->getDavixFileInstance()) == NULL)
       return kTRUE;
-   Long64_t ret = DavixReadBuffer(fd, buf, len);
+   Long64_t ret = DavixReadBuffer(fd, buf, fOffset, len);
+   fOffset += std::max((Long64_t)0, ret); // only shift offset on successful read
```

Comment:
Doesn't this operation also need the lock?

Reply:
Right, I was thinking about the modification of the offset data member, in case another thread is reading it from another method of the class at the same time.

Reply:
>  I was thinking about the modification of the offset data member

It is currently 'not allowed' (i.e. explicitly not thread safe) to use the same `TFile` instance from multiple threads.

Reply:
Yes, that was also my assumption!